packages/cli/openapi-ir-to-fern/src/converters/convertEndpoint.ts

```diff
@@ -289,34 +289,53 @@ function convertNamedFullExamplesToExampleTypeReferenceSchemas(
 function convertFullExampleToExampleTypeReferenceSchema(
     fullExample: FullExample
 ): RawSchemas.ExampleTypeReferenceSchema {
-    return FullExample._visit<RawSchemas.ExampleTypeReferenceSchema>(fullExample, {
-        primitive: (primitive) => convertPrimitiveExampleToExampleTypeReferenceSchema(primitive),
-        object: (object) => convertFullExamplePropertiesToExampleTypeReferenceSchema(object.properties),
-        array: (array) => convertFullArrayExampleToExampleTypeReferenceSchema(array),
-        map: (map) => convertFullMapExampleToExampleTypeReferenceSchema(map),
-        oneOf: (oneOf) => convertFullOneOfExampleToExampleTypeReferenceSchema(oneOf),
-        enum: (value) => value,
-        literal: (value) => value,
-        unknown: (unknown) => convertFullExampleToExampleTypeReferenceSchema(unknown),
-        _unknown: () => "unknown",
-    });
+    switch (fullExample.type) {
+        case "primitive":
+            return convertPrimitiveExampleToExampleTypeReferenceSchema(fullExample.primitive);
+        case "object":
+            return convertFullExamplePropertiesToExampleTypeReferenceSchema(fullExample.properties);
+        case "array":
+            return convertFullArrayExampleToExampleTypeReferenceSchema(fullExample.array);
+        case "map":
+            return convertFullMapExampleToExampleTypeReferenceSchema(fullExample.map);
+        case "oneOf":
+            return convertFullOneOfExampleToExampleTypeReferenceSchema(fullExample.oneOf);
+        case "enum":
+            return fullExample.enum;
+        case "literal":
+            return fullExample.literal;
+        case "unknown":
+            return convertFullExampleToExampleTypeReferenceSchema(fullExample.unknown);
+        default:
+            throw new Error("Cannot convert unsupported example type");
```

Comment:
i would do `assertNever(fullExample)` the ts compiler will make sure that the code never gets there

Reply:
Nice, good call 👍